core/storage/classifier/gae_models.py

```diff
@@ -111,3 +113,90 @@ def create(
 
         classifier_model_instance.put()
         return instance_id
+
+
+class ClassifierTrainingJobModel(base_models.BaseModel):
+    """Model for storing classifier training jobs.
+
+    The id of instances of this class has the form
+    {{exp_id}}.{{random_hash_of_16_chars}}
+    """
+
+    # The ID of the algorithm used to create the model.
+    algorithm_id = ndb.StringProperty(required=True, choices=ALGORITHM_CHOICES)
+    # The exploration_id of the exploration to whose state the model belongs.
+    exp_id = ndb.StringProperty(required=True)
+    # The exploration version at the time this training job was created.
+    exp_version_when_created = ndb.IntegerProperty(required=True)
+    # The name of the state to which the model belongs.
+    state_name = ndb.StringProperty(required=True)
+    # The status of the training job. It can be either NEW, COMPLETE or PENDING
+    status = ndb.StringProperty(required=True,
```

Comment:
But it will also slow down the inserts, updates and deletes. And indices do no make anything "easy" instead they make only the select queries faster (and the speed up depends on the data). We have just one read query vs one insert (when the job is created), two updates (when status is updated). And, indices are updated for every insert/delete/update hence making them unsuitable for anything other than "read" heavy processes. So it is really about carefully considering the use case instead of just using the indices.